core_helpers/logging_objects.py

```diff
@@ -83,3 +95,49 @@ def failure(self,msg,level='info',pre_format='FAIL: '):
     def set_rp_logger(self,rp_pytest_service):
         "Set the reportportal logger"
         self.rp_logger = self.log_obj.setup_rp_logging(rp_pytest_service)
+
+    def conditional_write(self,flag,positive,negative,level='info'):
+        "Write out either the positive or the negative message based on flag"
+        self.mini_check_counter += 1
+        if level.lower() == "inverse":
+            if flag is True:
+                self.write(positive,level='error')
+            else:
+                self.write(negative,level='info')
+                self.mini_check_pass_counter += 1
+        else:
+            if flag is True:
+                self.write(positive,level='info')
+                self.mini_check_pass_counter += 1
+            else:
+                self.write(negative,level='error')
+
+    def make_gif(self):
```

Comment:
Do we need this method in `Logging_Objects`?

Reply:
I had moved the `make_gif` method into logging objects as it is called in the `write_test_summary`. Sice we are inheriting only the logging objects class in the API player, it is would be unable to use the `make_gif` if it was in the `screenshot_objects`.

Reply:
Okay. Did you check if we use `make_gif` in `API` framework?

Reply:
We don't use `make_gif` in the API.  I initially tried setting `gif_file_name` to None on initialization so that we could skip the `make_gif` method in `write_test_summary` for API tests, however the `gif_file_name` itself is being set inside the `make_gif` method.
So, I have updated this PR by modified the `write_test_summary `method to execute the `make_gif` method , only if `screenshot_dir` is not None, that way it would skip it for API tests